src/main/java/io/supertokens/jwt/JWTSigningFunctions.java

```diff
@@ -56,69 +85,80 @@ public static String createJWTToken(Main main, String algorithm, JsonObject payl
     public static List<JsonObject> getJWKS(Main main)
             throws StorageQueryException, StorageTransactionLogicException, NoSuchAlgorithmException,
             InvalidKeySpecException {
+        // Retrieve all keys in storage
         List<JWTSigningKeyInfo> keys = JWTSigningKey.getInstance(main).getAllSigningKeys();
         List<JsonObject> jwks = new ArrayList<>();
 
         for (int i = 0; i < keys.size(); i++) {
             JWTSigningKeyInfo currentKeyInfo = keys.get(i);
-            RSAPublicKey publicKey = getPublicKeyFromString(currentKeyInfo.publicKey, currentKeyInfo.algorithmType.toUpperCase());
-            JsonObject jwk = new JsonObject();
 
-            jwk.addProperty("kty", currentKeyInfo.algorithmType);
-            jwk.addProperty("kid", currentKeyInfo.keyId);
-            jwk.addProperty("n", Base64.getUrlEncoder().encodeToString(publicKey.getModulus().toByteArray()));
-            jwk.addProperty("e", Base64.getUrlEncoder().encodeToString(publicKey.getPublicExponent().toByteArray()));
-            jwk.addProperty("alg", currentKeyInfo.algorithm.toUpperCase());
-            jwk.addProperty("use", "sig");
-
-            jwks.add(jwk);
+            // We only use asymmetric keys
+            if (currentKeyInfo instanceof JWTAsymmetricSigningKeyInfo) {
+                JWTSigningKey.SupportedAlgorithms algorithm = JWTSigningKey.SupportedAlgorithms.valueOf(currentKeyInfo.algorithm);
+                // TODO: In the future with more asymmetric algorithms [ES256 for example] we will need a provider system for the public key + JWK - Nemi
+                RSAPublicKey publicKey = getPublicKeyFromString(( (JWTAsymmetricSigningKeyInfo) currentKeyInfo).publicKey, algorithm);
+                JsonObject jwk = new JsonObject();
+
+                jwk.addProperty("kty", algorithm.getAlgorithmType());
+                jwk.addProperty("kid", currentKeyInfo.keyId);
+                jwk.addProperty("n", Base64.getUrlEncoder().encodeToString(publicKey.getModulus().toByteArray()));
+                jwk.addProperty("e", Base64.getUrlEncoder().encodeToString(publicKey.getPublicExponent().toByteArray()));
+                jwk.addProperty("alg", currentKeyInfo.algorithm.toUpperCase());
+                jwk.addProperty("use", "sig");
+
+                jwks.add(jwk);
+            }
         }
 
         return jwks;
     }
 
-    private static JWTSigningKeyInfo getKeyToUse(Main main, String algorithm)
-            throws StorageQueryException, StorageTransactionLogicException, NoSuchAlgorithmException {
-        return JWTSigningKey.getInstance(main).getKeyForAlgorithm(algorithm);
-    }
-
-    private static Algorithm getAlgorithmFromString(Main main, String algorithm, JWTSigningKeyInfo keyToUse) throws NoSuchAlgorithmException, InvalidKeySpecException {
+    private static Algorithm getAuth0Algorithm(JWTSigningKey.SupportedAlgorithms algorithm, JWTSigningKeyInfo keyToUse) throws NoSuchAlgorithmException, InvalidKeySpecException,
```

Comment:
Its directly using Auth0s package which calls the object "Algorithm" which I thought could be confusing since that is used for the signing algorithm as well, for the lack of a better name I used this one